### DISCUSSION TITLE
mapbox-gl jump-to/flyTo option for react-map-gl

### DISCUSSION THREAD

**Discussion by emekaokoli** [uber/react-map-gl#1959] [Q&A]
Please is there something like this in react-map-gl

```
map.on('load', () => {
for (const [index, coordinate] of cityCoordinates.entries()) {
map.jumpTo({ center: coordinate });
}
});

```
or 

```
map.on('load', () => {
map.flyTo({
center: [122333, 02432222],
essential: true // this animation is considered essential with respect to prefers-reduced-motion
});
```

I am looking to dynamically move the map to the marker point when the coordinates are supplied from the fetched response data.

**Comment by Pessimistress** [comment]
https://github.com/visgl/react-map-gl/blob/7.0-release/examples/viewport-animation/src/app.tsx#L22

**Reply by Pessimistress** [reply]
Search is your friend.

![Screen Shot 2022-08-23 at 4 36 47 PM](https://user-images.githubusercontent.com/2059298/186284495-5c703acb-f169-4de9-aee5-c186b945005e.png)

**Comment by emekaokoli** [comment]
I found it and implemented it but it doesn’t fly to onload , I also tried
mapref.current.setcenter() still not working.

On Wed, 24 Aug 2022 at 12:37 AM, Xiaoji Chen ***@***.***>
wrote:

> Search is your friend.
>
> [image: Screen Shot 2022-08-23 at 4 36 47 PM]
> <https://user-images.githubusercontent.com/2059298/186284495-5c703acb-f169-4de9-aee5-c186b945005e.png>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/visgl/react-map-gl/discussions/1959#discussioncomment-3460930>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AANGI5IJJXUWHX5JKKLCL5LV2VOE5ANCNFSM57NHBW4Q>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
-- 
*OKOLI emeka,*
*Full-stack Software Engineer, *
*Enugu, **Nigeria*
***@***.*** ***@***.***>*
*LinkedIn: http://www.linkedin.com/in/emekaokoli/
<http://www.linkedin.com/in/emekaokoli/>*
*Github: https://github.com/emekaokoli <https://github.com/emekaokoli>*

**Reply by Pessimistress** [reply]
Create a CodeSandbox if you need help.

**Reply by emekaokoli** [reply]
here it is

https://codesandbox.io/s/headless-cache-ccjs96?file=/src/map.js:2190-2197

**Reply by Pessimistress** [reply]
```js
const onMapLoad = useCallback(() => {
      mapRef.current.flyTo({
        center: [lng, lat],
        duration: 2000
       });
  }, [lng, lat]);
```

`onMapLoad` is already called on the `load` event. You register a new load event listener inside `onMapLoad`. It will never get called because load only fires once.

**Reply by emekaokoli** [reply]
still does not work, I have to manually pan/move the map to find the location of the marker's new long/Lat.

**Reply by emekaokoli** [reply]
Solved it with this.

```
  useEffect(() => {
    if (mapRef.current) {
      property.map(({ laty_decimal, longx_decimal }) =>
        mapRef.current?.setCenter({
          lat: laty_decimal,
          lng: longx_decimal,
        })
      );
    }
  }, [property]);
```

thanks for the help.
